app/src/main/java/org/lineageos/selfie/SettingsActivity.kt

```diff
@@ -75,6 +80,29 @@ class SettingsActivity : AppCompatActivity() {
                     }
             }
             shutterSound?.isEnabled = !CameraSoundsUtils.mustPlaySounds
+            videoQuality?.let {
+                val supportedVideoQualities = supportedVideoQualities()
+                val map = it.entries.zip(it.entryValues).toMap().filter { (_, value) ->
+                    value == "lowest" || value == "highest" || supportedVideoQualities.contains(
+                        when (value) {
+                            "sd" -> Quality.SD
+                            "hd" -> Quality.HD
+                            "fhd" -> Quality.FHD
+                            "uhd" -> Quality.UHD
+                            else -> null
+                        }
+                    )
+                }
+                it.entries = map.keys.toTypedArray()
+                it.entryValues = map.values.toTypedArray()
+            }
+        }
+
+        @androidx.camera.camera2.interop.ExperimentalCamera2Interop
+        private fun supportedVideoQualities(): List<Quality> {
+            val cameraProvider = ProcessCameraProvider.getInstance(requireContext()).get()
+            val cameraInfo = cameraProvider.availableCameraInfos[0]
```

Comment:
Every camera has different video qualities

Reply:
Yes, aware of that. I suppose we can reuse this later to add separate back facing camera / front facing camera video quality pref.